src/packet/buffer_rx.rs

```diff
@@ -146,49 +148,132 @@ impl DepacketizingBuffer {
         //     self.segments
         // );
 
-        let (start, stop) = self.segments.first()?;
+        let (start, stop) = *self.segments.first()?;
+
+        let seq = {
+            let last = self.queue.get(stop).expect("entry for stop index");
+            last.meta.seq_no
+        };
+
+        // depack ahead to check contiguity, even if we may not emit right away
+        let dep = match self.depacketize(start, stop, seq) {
+            Ok(d) => d,
+            Err(e) => {
+                // this segment cannot be decoded correctly
+                // remove from the queue and return the error
+                self.last_emitted = Some((seq, CodecExtra::None));
+                self.queue.drain(0..=stop);
+                return Some(Err(e));
+            }
+        };
+
+        let contiguous = 'contig: {
+            if self.is_following_last(start) {
+                break 'contig true;
+            }
+
+            let Some((last_seq, last_codec_extra)) = self.last_emitted else {
+                break 'contig true;
+            };
+
+            const VP8_DEPACK_OPTIMIZATION: bool = true;
+
+            match (last_codec_extra, dep.codec_extra) {
+                (CodecExtra::Vp8(prev), CodecExtra::Vp8(next)) => {
+                    if !VP8_DEPACK_OPTIMIZATION {
+                        break 'contig false;
+                    }
+
+                    // In the case of VP8 chrome does answer nacks for frames that are on
+                    // temporal layer1 Since VP8 frames are interleaved, we can tolerate a
+                    // missing frame on layer 1 that its contiguous to two frames on layer 0
+
+                    let Some(prev_pid) = prev.picture_id else {
+                        break 'contig false;
+                    };
+                    let Some(next_pid) = next.picture_id else {
+                        break 'contig false;
+                    };
+
+                    let allowed = prev.layer_index == 0
```

Comment:
This logic is only correct if we have two temporal layers, right? If we had a single temporal layer we'd be missing a picture from it here

Reply:
Also we assume that VP8 frames arrive in order always, is that specified in the RTP VP8 RFC?

Reply:
What happens if we get a subsequent L1 frame here i.e.

```
F0(L0), F2(L0), F3(L1), F4(L0)
```

The consumer will see this as contiguous despite F1 being missing. Presumably the consumer should not use F3 here?

Reply:
yes, we only have 2 temporal layer in webrtc. Chrome has experimented 4 but then reverted to 2. I don't remember where this is documented.

I don't think we assume they arrive in order. Our segments are in order.

As far as we know vp8 layers are interleaved L0, L1, L0, L1, .. 

Is the missing F1 on Layer 1 ? In that case the decoder (should) understand that F3 depends on a missing F1 and therefore skip it. If this is not the case, it just means that we'd have to suppress L1 frames until the next L1 marked as sync passes through. Right now latency has improved and the stream is mostly ok considering the amount of loss. I did see minor glitches (but very hard to see and repro) so I may investigate this more when I am able to reproduce better.

Reply:
> I don't think we assume they arrive in order. Our segments are in order.

I mean after we have reordered them from RTP, are the VP8 frames in order then?

> Is the missing F1 on Layer 1 ? In that case the decoder (should) understand that F3 depends on a missing F1 and therefore skip it. If this is not the case, it just means that we'd have to suppress L1 frames until the next L1 marked as sync passes through. Right now latency has improved and the stream is mostly ok considering the amount of loss I simulate. I did see minor glitches (but very hard to see and repro) so I may investigate this more when I am able to reproduce better.

Cool, that's what I imagined